scheduler/reconcile.go

```diff
@@ -850,6 +861,11 @@ func (a *allocReconciler) handleDelayedReschedules(rescheduleLater []*delayedRes
 	evals = append(evals, eval)
 
 	for _, allocReschedInfo := range rescheduleLater {
+		a.result.stop = append(a.result.stop, allocStopResult{
+			alloc:             allocReschedInfo.alloc,
+			statusDescription: allocRescheduled,
+		})
```

Comment:
Here I'm opting to mark allocations that are due to be re-shceduled later to stop immediately.  At this point, the alloc already failed enough times such that client wouldn't restart it, so I decided to stop alloc immediately rather than wait until the follow up alloc is placed.

Reply:
This seems incorrect - it will cause the same alloc to be both considered Ignored and Stopped, from what I see in local testing. Lets talk about this offline. 

Marking it as stopped when the replacement alloc is created works as expected.

Reply:
After digging into it, I see that this behavior is indeed incorrect.  We cause the allocs to be marked as Stop twice: once when it's recognized as rescheduledLater, and once again when we process the rescheduleLater in the follow up eval.

Removed this rescheduleLater in 34a66835db20eea7d5558c340f3f7b7a0d2b6ca7 with a commit message explaining the rationale.